test/e2e/utils_test.go

```diff
@@ -564,3 +568,130 @@ func waitForIngressReadiness(ctx context.Context, client kubernetes.Interface, i
+// once the job is completed, otherwise should return a time-out error
+func pollTillJobCompleted(ctx context.Context, clientset *kubernetes.Clientset, namespace, jobName string) error {
+	err := wait.PollUntilContextTimeout(ctx, PollInterval, TestTimeout, true, func(ctx context.Context) (bool, error) {
+		job, err := clientset.BatchV1().Jobs(namespace).Get(ctx, jobName, metav1.GetOptions{})
+
+		if err != nil {
+			return false, err
+		}
+
+		for _, cond := range job.Status.Conditions {
+			if cond.Type == batchv1.JobComplete {
+				if cond.Status == corev1.ConditionTrue {
+					return true, nil
+				} else {
+					return false, nil
+				}
+			}
+		}
+
+		return false, nil
+	})
+	return err
+}
+
+// pollTillServiceAccountAvailable poll the service account object and returns non-nil error
+// once the service account is available, otherwise should return a time-out error
+func pollTillServiceAccountAvailable(ctx context.Context, clientset *kubernetes.Clientset, namespace, serviceAccountName string) error {
+	err := wait.PollUntilContextTimeout(ctx, PollInterval, TestTimeout, true, func(ctx context.Context) (bool, error) {
+		sa, err := clientset.CoreV1().ServiceAccounts(namespace).Get(ctx, serviceAccountName, metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
+		}
+
+		return sa != nil, nil
+	})
+
+	return err
+}
+
+// pollTillIstioCSRAvailable poll the istioCSR object and returns non-nil error and istio-grpc-endpoint
+// once the istiocsr is available, otherwise should return a time-out error
+func pollTillIstioCSRAvailable(ctx context.Context, dynamicClient *dynamic.DynamicClient, namespace, istioCsrName string) (string, error) {
+	var istioCSRGRPCEndpoint string
+	err := wait.PollUntilContextTimeout(ctx, PollInterval, TestTimeout, true, func(ctx context.Context) (bool, error) {
+		gvr := schema.GroupVersionResource{
+			Group:    "operator.openshift.io",
+			Version:  "v1alpha1",
+			Resource: "istiocsrs",
+		}
+
+		customResource, err := dynamicClient.Resource(gvr).Namespace(namespace).Get(ctx, istioCsrName, metav1.GetOptions{})
+		if err != nil {
+			return false, nil
+		}
+
+		status, found, err := unstructured.NestedMap(customResource.Object, "status")
+		if err != nil {
+			return false, nil
+		}
+
+		if !found {
+			return false, nil
+		}
+
+		conditions, found, err := unstructured.NestedSlice(customResource.Object, "status", "conditions")
+		if err != nil {
+			return false, nil
+		}
+
+		if !found {
+			return false, nil
+		}
+
+		for _, condition := range conditions {
+			condMap, ok := condition.(map[string]interface{})
+			if !ok {
+				continue
+			}
+
+			condType, _ := condMap["type"].(string)
+			condStatus, _ := condMap["status"].(string)
+
+			if condType != "Ready" {
+				continue
+			}
+
+			if condStatus == string(metav1.ConditionTrue) {
+				break
+			} else {
+				return false, nil
+			}
+
+		}
```

Comment:
please parse the status.conditions into a metav1.Condition, and use the https://pkg.go.dev/k8s.io/apimachinery/pkg/api/meta#FindStatusCondition helper func instead.